kbchat/kvstore_test.go

```diff
@@ -0,0 +1,60 @@
+package kbchat
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// test doesn't make assumption about prior state
+func TestKVStore(t *testing.T) {
+	alice, dir := testBotSetup(t, "alice")
+	channel := getTeamChatChannel(t, "acme")
+	team := channel.Name
+	defer testBotTeardown(t, alice, dir)
+
+	namespace := "_test_namespace1"
+	key := "_test_key1"
+
+	// put with default revision
+	res2, err2 := alice.PutEntry(team, namespace, key, "value1", 0)
+	require.NoError(t, err2)
+	rev := res2.Revision
+
+	// fail put
+	_, err3 := alice.PutEntry(team, namespace, key, "value2", rev)
+	require.Error(t, err3)
+
+	// list namespaces
+	res4, err4 := alice.ListNamespaces(team)
+	require.NoError(t, err4)
+	require.True(t, len(res4.Namespaces) > 0)
+
+	// list entryKeys
+	res5, err5 := alice.ListEntryKeys(team, namespace)
+	require.NoError(t, err5)
+	require.True(t, len(res5.EntryKeys) > 0)
+
+	// get
+	res6, err6 := alice.GetEntry(team, namespace, key)
+	require.NoError(t, err6)
+	require.Equal(t, "value1", res6.EntryValue)
+
+	// fail delete
+	_, err7 := alice.DeleteEntry(team, namespace, key, rev+2)
```

Comment:
i suggest incrementing `rev` as you work your way down your test when things work. this will make it easier for someone to slip something in (or remove something) in the middle of this.

Reply:
well, this one is supposed to fail, but the next one